model/utils.go

```diff
@@ -575,73 +512,24 @@ func IsValidId(value string) bool {
 	return true
 }
 
-// Copied from https://golang.org/src/net/dnsclient.go#L119
-func IsDomainName(s string) bool {
-	// See RFC 1035, RFC 3696.
-	// Presentation format has dots before every label except the first, and the
-	// terminal empty label is optional here because we assume fully-qualified
-	// (absolute) input. We must therefore reserve space for the first and last
-	// labels' length octets in wire format, where they are necessary and the
-	// maximum total length is 255.
-	// So our _effective_ maximum is 253, but 254 is not rejected if the last
-	// character is a dot.
-	l := len(s)
-	if l == 0 || l > 254 || l == 254 && s[l-1] != '.' {
-		return false
-	}
-
-	last := byte('.')
-	ok := false // Ok once we've seen a letter.
-	partlen := 0
-	for i := 0; i < len(s); i++ {
-		c := s[i]
-		switch {
-		default:
-			return false
-		case 'a' <= c && c <= 'z' || 'A' <= c && c <= 'Z' || c == '_':
-			ok = true
-			partlen++
-		case '0' <= c && c <= '9':
-			// fine
-			partlen++
-		case c == '-':
-			// Byte before dash cannot be dot.
-			if last == '.' {
-				return false
-			}
-			partlen++
-		case c == '.':
-			// Byte before dot cannot be dot, dash.
-			if last == '.' || last == '-' {
-				return false
-			}
-			if partlen > 63 || partlen == 0 {
-				return false
-			}
-			partlen = 0
-		}
-		last = c
-	}
-	if last == '-' || partlen > 63 {
-		return false
+// RemoveDuplicateStrings does an in-place removal of duplicate strings
+// from the input slice.
+func RemoveDuplicateStrings(in []string) {
+	// In-place de-dup.
+	// Copied from https://github.com/golang/go/wiki/SliceTricks#in-place-deduplicate-comparable
+	if len(in) == 0 {
+		return
 	}
-
-	return ok
-}
-
-func RemoveDuplicateStrings(in []string) []string {
-	out := []string{}
-	seen := make(map[string]bool, len(in))
-
-	for _, item := range in {
-		if !seen[item] {
-			out = append(out, item)
-
-			seen[item] = true
+	sort.Strings(in)
+	j := 0
+	for i := 1; i < len(in); i++ {
+		if in[j] == in[i] {
+			continue
 		}
+		j++
+		in[j] = in[i]
 	}
-
-	return out
+	in = in[:j+1] //nolint:staticcheck
```

Comment:
Not sure this is going work actually :/ I think we cannot propagate the change in length since we are technically handling a copy of the given slice (although sharing the same backing data).

Reply:
Ahh you are right actually. So basically we will need to document the changes. And just mention that the returned slice is reused.

Reply:
Indeed.